test/val/val_id_test.cpp

```diff
@@ -1925,6 +1925,169 @@ TEST_F(ValidateIdWithMessage, OpFunctionCallArgumentTypeBad) {
+          %3             = OpTypeFunction %void
+      %float             = OpTypeFloat 32
+    %v4float             = OpTypeVector %float 4
+%_ptr_Function_v4float   = OpTypePointer Function %v4float
+         %10             = OpTypeImage %float 2D 0 0 0 1 Unknown
+%_ptr_UniformConstant_10 = OpTypePointer UniformConstant %10
+        %tex             = OpVariable %_ptr_UniformConstant_10 UniformConstant
+         %14             = OpTypeSampler
+%_ptr_UniformConstant_14 = OpTypePointer UniformConstant %14
+          %s             = OpVariable %_ptr_UniformConstant_14 UniformConstant
+         %18             = OpTypeSampledImage %10
+    %v2float             = OpTypeVector %float 2
+    %float_1             = OpConstant %float 1
+         %23             = OpConstantComposite %v2float %float_1 %float_1
+       %main             = OpFunction %void None %3
+          %5             = OpLabel
+          %x             = OpVariable %_ptr_Function_v4float Function
+         %13             = OpLoad %10 %tex
+         %17             = OpLoad %14 %s
+         %19             = OpSampledImage %18 %13 %17
+         %24             = OpImageSampleImplicitLod %v4float %19 %23
+         OpStore %x %24
+         OpReturn
+         OpFunctionEnd)";
+  CompileSuccessfully(spirv.c_str());
+  EXPECT_EQ(SPV_SUCCESS, ValidateInstructions());
+}
+
+// Invalid: OpSampledImage result <id> is defined in one block and used in a
+// different block.
+TEST_F(ValidateIdWithMessage, OpSampledImageUsedInDifferentBlockBad) {
+  string spirv = kGLSL450MemoryModel + R"(
+       %void             = OpTypeVoid
+          %3             = OpTypeFunction %void
+      %float             = OpTypeFloat 32
+    %v4float             = OpTypeVector %float 4
+%_ptr_Function_v4float   = OpTypePointer Function %v4float
+         %10             = OpTypeImage %float 2D 0 0 0 1 Unknown
+%_ptr_UniformConstant_10 = OpTypePointer UniformConstant %10
+        %tex             = OpVariable %_ptr_UniformConstant_10 UniformConstant
+         %14             = OpTypeSampler
+%_ptr_UniformConstant_14 = OpTypePointer UniformConstant %14
+          %s             = OpVariable %_ptr_UniformConstant_14 UniformConstant
+         %18             = OpTypeSampledImage %10
+    %v2float             = OpTypeVector %float 2
+    %float_1             = OpConstant %float 1
+    %float_2             = OpConstant %float 2
+         %23             = OpConstantComposite %v2float %float_1 %float_1
+         %24             = OpConstantComposite %v2float %float_2 %float_2
+         %40             = OpTypeBool
+         %41             = OpSpecConstantTrue %40
+       %main             = OpFunction %void None %3
+          %5             = OpLabel
+          %x             = OpVariable %_ptr_Function_v4float Function
+         %13             = OpLoad %10 %tex
+         %17             = OpLoad %14 %s
+         %25             = OpSampledImage %18 %13 %17
+         OpBranchConditional %41 %50 %70
```

Comment:
This will fail validation in Shader due to structured control flow rules.  You're missing an OpSelectionMerge.

Much better still is to use two basic blocks, and an unconditional branch from the first to the second.